src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,242 @@
+import classNames from 'classnames';
+import React, {
+  Dispatch, SetStateAction, useState,
+} from 'react';
+import { Post } from '../../types/Post';
+import { Comment } from '../../types/Comment';
+
+type Props = {
+  selectedPost: Post,
+  comments: Comment[],
+  onAdd: (comment: Comment) => void;
+  loading: boolean,
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  comments,
+  onAdd,
+  loading,
+}) => {
+  const [name, setName] = useState('');
+  const [nameVisited, setNameVisited] = useState(false);
+
+  const [email, setEmail] = useState('');
+  const [emailVisited, setEmailVisited] = useState(false);
+
+  const [body, setBody] = useState('');
+  const [commentVisited, setCommentVisited] = useState(false);
+
+  const [newCommentId, setNewCommentId] = useState(0);
+
+  const handleBlur = (
+    fieldValue: string,
+    setFieldVisited: Dispatch<SetStateAction<boolean>>,
+  ) => {
+    if (!fieldValue.trim()) {
+      setFieldVisited(true);
+    }
+  };
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement>
+    | React.ChangeEvent<HTMLTextAreaElement>,
+    setFieldValue: Dispatch<SetStateAction<string>>,
+    setFieldVisited: Dispatch<SetStateAction<boolean>>,
+  ) => {
+    setFieldVisited(true);
+    setFieldValue(e.target.value);
+  };
+
+  const handleError = (
+    fieldValue: string,
+    fieldVisited: boolean,
+  ) => {
+    return ((fieldValue.length === 0) && fieldVisited);
+  };
+
+  const resetError = (
+    setFieldValue: Dispatch<SetStateAction<string>>,
+    setFieldVisited: Dispatch<SetStateAction<boolean>>,
+  ) => {
+    setFieldValue('');
+    setFieldVisited(false);
+  };
+
+  const handleNewCommentId = (postId: number) => {
+    const postComments = comments.filter(note => note.postId === postId);
+    const maxCommentId = Math.max(...postComments.map(el => el.id));
+
+    setNewCommentId(maxCommentId + 1);
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    handleNewCommentId(selectedPost.id);
```

Comment:
Consider moving the 'handleNewCommentId' function call after the form validation checks. If the form is invalid, there's no need to generate a new comment id.